extensions/extensions-helm/extensions-helm-spark-infrastructure/aissemble-hive-metastore-service-chart/templates/configmap.yaml

```diff
@@ -8,20 +8,22 @@ data:
   # Add all the default properties from the local values.yaml to the ConfigMap
   # Then check if there are any downstream properties and add them as well
   metastore-site.xml: |
-    {{- $basePropDict := dict }}
-    {{- $propDict := dict }}
-    {{- range $baseProperty := .Values.configMap.metastoreServiceConfig.baseProperties }}
-      {{- $basePropDict := set $basePropDict $baseProperty.name $baseProperty.value }}
+    {{- define "hasProperty"}}
+    {{- range $property := $.Values.configMap.metastoreServiceConfig.properties }}
+      {{- if eq $property.name $.baseProp.name }}  
+        {{ "true" }}
+        {{- break }}
+      {{- else }}
+        {{"false"}}
+      {{- end }}  
     {{- end }}
-    {{- if .Values.configMap.metastoreServiceConfig.properties }}
-      {{- range $property := .Values.configMap.metastoreServiceConfig.properties }}
-        {{- $propDict := set $propDict $property.name $property.value }}
-      {{- end }}
     {{- end }}
     
     <configuration>
     {{- range $baseProperty := .Values.configMap.metastoreServiceConfig.baseProperties }}
-      {{- if and (hasKey $basePropDict $baseProperty.name) (not (hasKey $propDict $baseProperty.name))}}
+    {{- if $.Values.configMap.metastoreServiceConfig.properties }}
+      {{- $val :=  include "hasProperty" (dict "Values" $.Values "baseProp" $baseProperty) }}
```

Comment:
Q: Can you not use `hasKey` directly?

Reply:
Do you mean Create dictonary called propertiesDict and  (hasKey propertiesDict $baseProperty.name) ?

Directly using hasKey on .Values.configMap.metastoreServiceConfig.baseProperties doesn't work as this is list of maps (i.e. []map)